### ISSUE TITLE
js config is treated as yaml

### ISSUE THREAD

**Issue by ikatyang** [davidtheclark/cosmiconfig#95]
Context: https://github.com/prettier/prettier/issues/2908#issuecomment-332799662

For example, `module.exports = { foo: true }` is parsed as `{ "module.exports = { foo": "true }" }` while passing with its filename (`myConfig.js`), cosmiconfig should infer `format` based on its filename if there is no `format` provided instead of always `tryAllParsing`, which always try yaml parsing first.

something like:

https://github.com/davidtheclark/cosmiconfig/blob/ad33e75766c194eaa9f28df1d9688b3ce5a8e462/src/loadDefinedFile.js#L22

```js
const format = options.format ||
  /\.(js)$/.test(filepath)
    ? "js"
    : /\.(json)$/.test(filepath)
      ? "json"
      : /\.(yml|yaml)$/.test(filepath) ? "yaml" : undefined;

switch (format) {
```

~~I originally want to send a PR directly but failed since there are too many `expect error` tests failed I have no idea how to fix them.~~

**Comment by davidtheclark** [comment]
> I originally want to send a PR directly but failed since there are too many expect error tests failed I have no idea how to fix them.

@ikatyang I don't think I understand what this means. The tests were failing *after* your change? Or before?

**Comment by ikatyang** [comment]
The `failed-files.test.js` failed due to the different error message, I've opened #96 to fix this issue.

**Comment by sudo-suhas** [comment]
Fixed and released in `3.1.0` :tada:
